### PR TITLE
mypy: include mypy testing for the repository

### PR DESCRIPTION
**Pull request by lavocatt**
Run mypy for all files under tools/ and test/cases/
Fix the type errors in the files.

### CODE REVIEW

**Review by achilleas-k — APPROVED** [review]

File: tools/osbuild-image-test [hunk 1]
```diff
@@ -41,16 +41,16 @@ class OSBuild:
 
     def compile(self, manifest: dict, exports: List, checkpoints=None):
```
Comment (supakeen) [inline tools/osbuild-image-test]: Using the lowercase `dict` here is a 3.9+ only feature and likely doesn't do what we want < 3.9.

**Review by supakeen — CHANGES_REQUESTED** [review]
This all looks good. Could you get around the # type-ignore by passing --ignore-missing-imports to mypy or is that not a thing? The only thing I saw is the lowercase dict.

File: tools/osbuild-image-test (hunk 1 above)
Comment (lavocatt) [reply tools/osbuild-image-test]: Done :) Thanks!

**Comment by lavocatt** [comment]
> This all looks good. Could you get around the # type-ignore by passing --ignore-missing-imports to mypy or is that not a thing? The only thing I saw is the lowercase dict.

Indeed that was better with this option.

File: tools/osbuild-image-test [hunk 2]
```diff
@@ -321,9 +326,9 @@ class TestCases:
 
     @staticmethod
     def _print_stats(tests: List[TestCase]) -> None:
-        distros = defaultdict(int)
-        archs = defaultdict(int)
-        types = defaultdict(int)
+        distros: Dict[str, Any] = defaultdict(int)
+        archs: Dict[str, Any] = defaultdict(int)
+        types: Dict[str, Any] = defaultdict(int)
```
Comment (supakeen) [inline tools/osbuild-image-test]: Wouldn't the `defaultdict(int)` here indicate that it should be `Dict[str, int]`? :)

File: tools/osbuild-image-test [hunk 3]
```diff
@@ -39,18 +39,18 @@ class OSBuild:
...
+                        checkpoints=checkpoints)
         return res
 
     def _run(self,
-            args: Optional[List[str]] = None,
-            manifest: Optional[dict] = None,
-            exports: Optional[List] = None,
-            checkpoints: Optional[List] = None):
+             args: Optional[List[str]] = None,
+             manifest: Optional[dict] = None,
```
Comment (supakeen) [inline tools/osbuild-image-test:51]: One last lowercased `dict`.

**Review by supakeen — COMMENTED** [review]
Two tiny things I spotted.

**Review by supakeen — APPROVED** [review]

File: tools/osbuild-image-test (hunk 2 above)
Comment (lavocatt) [reply tools/osbuild-image-test]: :D thanks!

**Pull request merged**